.github/workflows/test.yml

```diff
@@ -113,13 +113,6 @@ jobs:
           ghc: ${{ matrix.ghc }}
           os:  ${{ runner.os }}
 
-      # repeating builds to workaround segfaults in windows and ghc-8.8.4
-      - name: Build
```

Comment:
Previously we could have had plugins which _built_ on GHC X (and hence weren't commented out in the special `cabal.project`), but whose _tests_ didn't pass (and hence weren't included in the testing workflow. These plugins would have been built by this step, which would have ensured that they kept compiling, even if the tests didn't pass.

Now I think we will only build stuff that's necessary for the tests that we try and run, so if we don't test a plugin we also won't ensure that it keeps building.

Also I think we won't check compilation of the main HLS executable either?

Reply:
This is a good observation, thanks! 

I deleted this step before figuring out that we can use the `buildable` field to selectively disable projects per ghc version. With that trick, hopefully this step can stay and we can avoid all the pitfalls that you discovered

Reply:
I have a feeling that that won't work. I think I suggested this to @jneira in the past and he convinced me that it wouldn't work. I can't remember why though >:(

Reply:
I can't see the reason why it wouldn't work. CI is failing because the Cabal plan it computes is wrong for 9.0.1, but I get the right plan locally so it might be a CI bug

Reply:
I think maybe it had something to do with Cabal not taking `buildable` into account when making a plan, so it won't be clever enough to try setting the `splice` flag to `false` if the splice plugin isn't buildable, it'll just fail. Worth checking though.

Reply:
It doesn't need to be:

https://github.com/haskell/haskell-language-server/blob/4b475b0adf665d159b1de54f260eeb84bfdf58fe/haskell-language-server.cabal#L237-L239

Reply:
Ah, so you're going to keep the existing machinery? Maybe the thing that wouldn't work was a more ambitious thing: I'd though that maybe we could *automatically* have plugins be enabled or disabled depending on whether the plugin package was buildable given the rest of the configuration or not. So then we'd only have to put the logic for deciding buildability in the plugin package, which would be nice.

Reply:
One step at a time..